sbi/utils/posterior_ensemble.py

```diff
@@ -0,0 +1,444 @@
+                        learning_rate=learning_rate,
+                        init_method=init_method,
+                        num_init_samples=num_init_samples,
+                        save_best_every=save_best_every,
+                        show_progress_bars=show_progress_bars,
+                    )
+                )
+            maps = torch.stack(maps)
+            return log_weights, maps
+
+        else:
+            self.potential_fn.set_x(self._x_else_default_x(x))
+
+            if init_method == "posterior":
+                inits = self.sample((num_init_samples,), self._x_else_default_x(x))
+            elif init_method == "proposal":
+                inits = self.proposal.sample((num_init_samples,))
+            elif isinstance(init_method, Tensor):
+                inits = init_method
+            else:
+                raise ValueError
+
+            return gradient_ascent(
+                potential_fn=self.potential_fn,
+                inits=inits,
+                theta_transform=self.theta_transform,
+                num_iter=num_iter,
+                num_to_optimize=num_to_optimize,
+                learning_rate=learning_rate,
+                save_best_every=save_best_every,
+                show_progress_bars=show_progress_bars,
+            )[0]
+
+
+class EnsemblePotential(BasePotential):
+    r"""Provides `NeuralPosteriorEnsemble` with a `potential_fn` attribute.
+
+    The potential is the same as the sum of the weighted log-probabilities of each
+    component posterior.
+
+    This class was modelled off of `PosteriorBasedPotential` and should provide similar
+    functionality.
+
+    Attributes:
+        potential_fns: List of the potential_fns from each posterior component.
+        weights: Weights of each posterior distribution.
+    """
+
+    def __init__(
+        self,
+        potential_fns: List,
+        weights: Tensor,
+        priors: List[Any],
+        x_o: Optional[Tensor],
+        device: str = "cpu",
+    ):
+        r"""
+        Args:
+            potential_fns: List of the potential_fns from each posterior component.
+            weights: Weights of each posterior distribution.
+            priors: List of prior distributions.
+            x_o: Used as conditioning context for `potential_fn`.
+            device: Device which the component distributions sit on.
+        """
+        self._weights = weights
+        self.potential_fns = potential_fns
+
+        # list of priors potential_fn.prior
+        super().__init__(priors, x_o, device)
+
+        # self.set_x(x_o) # TODO: YES OR NO ? WILL IT BE DONE IN SUPER?
+
+    def allow_iid_x(self) -> bool:
+        if any(
+            isinstance(potential, PosteriorBasedPotential)
+            for potential in self.potential_fns
+        ):
+            # in case there is different kinds of posteriors, this will produce an error
+            # in `set_x()`
+            return False  # type: ignore
```

Comment:
why does this have a `#type: ignore`?

Reply:
I copied the line from a `PosteriorPotential` and forgot to remove it.